📖  Add vcluster to CAPI book Quick Start

oscr opened the pull request:
**What this PR does / why we need it**:

Adds a link to the vcluster provider Quick Start in the CAPI Quick Start.

killianmuldoon reviewed (approved):
Thank you!

/lgtm

fabriziopandini reviewed (changes requested):
From a quick look at the preview, there are at least two other tabs with provider-specific instructions on the quick start page, and ideally, all of them should have instructions for vcluster provider.
Could you kindly update them as well?

oscr commented on docs/book/src/user/quick-start.md:
```diff
@@ -562,7 +571,7 @@ before configuring a cluster with Cluster API. Instructions are provided for com
 discover the list of variables required by a cluster templates.
 
-{{#tabs name:"tab-configuration-infrastructure" tabs:"AWS,Azure,CloudStack,DigitalOcean,Docker,Equinix Metal,GCP,IBM Cloud,Metal3,Nutanix,Kubevirt,OpenStack,vSphere"}}
+{{#tabs name:"tab-configuration-infrastructure" tabs:"AWS,Azure,CloudStack,DigitalOcean,Docker,Equinix Metal,GCP,IBM Cloud,Metal3,Nutanix,Kubevirt,OpenStack,VCD,vcluster,vSphere"}}
```
I noticed VCD was missing so I added it.

oscr commented on docs/book/src/user/quick-start.md:
````diff
@@ -855,8 +864,20 @@ see [CAPVCD](https://github.com/vmware/cluster-api-provider-cloud-director)
 To see all required VCD environment variables execute:
 ```bash
 clusterctl generate cluster --infrastructure vcd --list-variables capi-quickstart
-``
````
Noticed a formatting mistake here. So added an additional ` to fix it

oscr commented:
I have updated with additional vcluster information. While doing so I noticed two mistakes that I fixed at the same time. If you prefer I can make a separate pr for them.

fabriziopandini commented on docs/book/src/user/quick-start.md:
```diff
@@ -897,7 +918,7 @@ For more information about prerequisites, credentials management, or permissions
 For the purpose of this tutorial, we'll name our cluster capi-quickstart.
 
-{{#tabs name:"tab-clusterctl-config-cluster" tabs:"Azure|AWS|CloudStack|DigitalOcean|Equinix Metal|GCP|Metal3|Nutanix|Kubevirt|OpenStack|vSphere,Docker"}}
+{{#tabs name:"tab-clusterctl-config-cluster" tabs:"Azure|AWS|CloudStack|DigitalOcean|Equinix Metal|GCP|Metal3|Nutanix|Kubevirt|OpenStack|vSphere,vcluster,Docker"}}
```
Do you mind to change this into
```suggestion
{{#tabs name:"tab-clusterctl-config-cluster" tabs:"Docker,vcluster,others..."}}
```
And fix tabs accordingly?
A tab so long looks really weird in the preview 
(otherwise I will follow up with another PR)

oscr replied:
No problem at all! The diff looks a little strange since I needed to reorder the tabs to get the correct content under the correct name.

fabriziopandini commented on docs/book/src/user/quick-start.md:
```diff
@@ -1010,7 +1047,7 @@ If you're using Docker Desktop on macOS, or Docker Desktop (Docker Engine works
 Calico is used here as an example.
 
-{{#tabs name:"tab-deploy-cni" tabs:"AWS|CloudStack|DigitalOcean|Docker|Equinix Metal|GCP|Metal3|Nutanix|Kubevirt|OpenStack|vSphere,Azure"}}
+{{#tabs name:"tab-deploy-cni" tabs:"AWS|CloudStack|DigitalOcean|Docker|Equinix Metal|GCP|Metal3|Nutanix|Kubevirt|OpenStack|vSphere,Azure,vcluster"}}
```
Same here
```suggestion
{{#tabs name:"tab-deploy-cni" tabs:"Azure,vcluster,others..."}}
```

fabriziopandini commented:
/lgtm
/approve

fabriziopandini commented:
/cherry-pick release-1.2
